pkg/tun/tun_linux.go

```diff
@@ -0,0 +1,153 @@
+// tun_linux.go: Open a Tunnel (L3 virtual interface) using the Universal TUN/TAP device driver.
+package tun
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/datawire/dlib/dexec"
+	"github.com/telepresenceio/telepresence/v2/pkg/tun/buffer"
+)
+
+const devicePath = "/dev/net/tun"
+
+type Device struct {
+	*os.File
+	name  string
+	index uint32
```

Comment:
Is there somewhere else that defines it as a `uint`? Because `<linux/if.h>` defines `ifru_index` as an `int`.

Reply:
`<linux/ipv6_route.h>` also has `rtmsg_ifindex` as an `int`.

Reply:
`int` is is, or rather `int32` since it's the go correspondance.